plugins/formal-verification/formver.core/src/org/jetbrains/kotlin/formver/conversion/StmtConversionVisitor.kt

```diff
@@ -112,7 +114,28 @@ class StmtConversionVisitor : FirVisitor<Exp, StmtConversionContext>() {
         val type = data.embedType(propertyAccessExpression)
         return when (symbol) {
             is FirValueParameterSymbol -> VariableEmbedding(symbol.callableId.embedName(), type).toLocalVar()
-            is FirPropertySymbol -> VariableEmbedding(symbol.callableId.embedName(), type).toLocalVar()
+
+            is FirPropertySymbol -> {
+                val varEmbedding = VariableEmbedding(symbol.callableId.embedName(), type)
+                if (symbol.isLocal) {
+                    return varEmbedding.toLocalVar()
+                } else {
+                    val receiver = propertyAccessExpression.dispatchReceiver.toResolvedCallableSymbol()!!
+                    val receiverType = receiver.resolvedReturnTypeRef.coneTypeOrNull!!
+                    val cvar = VariableEmbedding(receiver.callableId.embedName(), data.embedType(receiverType))
+                    val fieldAccess = Exp.FieldAccess(cvar.toLocalVar(), varEmbedding.toField())
```

Comment:
Instead of embedding the receiver yourself let the statement visitor do the work (that way it is also more general):
```kotlin
val receiver = propertyAccessExpression.dispatchReceiver.accept(this, data)
val fieldAccess = Exp.FieldAccess(receiver, varEmbedding.toField())
```